woocommerce/woocommerce-gutenberg-products-block
# Update Express Payments Loading UI

This PR makes some improvements to the Express Payments area and related components, notably:

- Split out the loading spinner from our button component so it can be used in the loading mask. This means we use one style of spinner animation instead of 2 :)
- Fixed LoadingMask to:
   - a) Block mouse interaction when masked
   - b) Not force children to re-render if loading state changes
- `useShallowEqual` now returns a default value instead of undefined. Updated typescript hints to match.
- Added loading mask to the express payments area
- Prevented excessive re-renders in express payments area by adding memorization to it's dependencies

Fixes #4159

### How to test the changes in this Pull Request:

To test this properly you'll need Stripe setup locally in sandbox mode. You can test the express payments with Chrome/Edge Pay, or Apple Pay in Safari.

1. Add an item to the cart
2. Go to checkout and wait for express payment methods to display
3. Toggle to change shipping; express payments should show the loading spinner and be blocked until the request is complete
4. Pay using an express payment method
5. After clicking Pay, the express payments area will show a loading spinner. When the request is complete and the checkout starts to redirect, the express payments area will remain blocked (no spinner).



### Changelog

> Show loading state in the express payments area whilst payment is processing or the page is redirecting.
